### ISSUE TITLE
`quotes` fixer can accidentally turn on strict mode when converting template backticks

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#7610]
**Tell us about your environment**

* **ESLint Version:** master
* **Node Version:** 7.1.0
* **npm Version:** 3.10.9

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```yml

rules:
  quotes: [error, single]

```

**What did you do? Please include the actual source code causing the issue.**

```js
// (I hope no one actually has code like this)

function foo() {
  `use strict`;
  return 012;
}

```

**What did you expect to happen?**

I expected `quotes` to report an error, but not fix it when using the `--fix` flag.

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint reported an error, and fixed the code to:

```js
function foo() {
  'use strict';
  return 012;
}
```

This is a SyntaxError, because octal literals are not allowed in strict mode.

Basically, the string `use strict` only turns on strict mode if it's surrounded by single or double quotes, not backticks. We should detect these cases and avoid performing a fix for them.

If someone does have `` `use strict` `` in backticks, they were probably trying to turn on strict mode and it would be nice if we could warn them that it wasn't working, but I think that's a separate issue. (edit: Actually, it looks like `no-unused-expressions` handles this correctly already.)
